### ISSUE TITLE
Cannot launch in wayland mode

### ISSUE THREAD

**Issue by red5h4d0w** [shiftkey/desktop#692]
### Search existing issues

- [X] I have searched the issue tracker and confirmed that I was unable to find existing issues for the problem I am experiencing.

### The problem

Cannot launch in native wayland mode using the usual flags. The app crashes after a SegFault when trying to do so.

### Release version

2.9.12RC4

### Operating system

EndeavourOS

### Steps to reproduce the behavior

0. Use a Wayland DE
1. Open a command prompt
2. run `github-desktop --enable-features=UseOzonePlatform --ozone-platform=wayland --verbose`

### Log files

```
./GitHubDesktop-linux-2.9.12-linux4.AppImage --verbose --enable-features=UseOzonePlatform --ozone-platform=wayland
[11818:0405/161655.624788:ERROR:gpu_init.cc(454)] Passthrough is not supported, GL is egl, ANGLE is 
[11818:0405/161655.639320:ERROR:sandbox_linux.cc(377)] InitializeSandbox() called with multiple threads in process gpu-process.
[11772:0405/161655.685004:ERROR:cursor_loader.cc(116)] Failed to load a platform cursor of type kNull
Erreur de segmentation (core dumped)

```

**Comment by shiftkey** [comment]
I don't have a working Wayland setup but this appears to be errors deep within Chromium:

```
./GitHubDesktop-linux-2.9.12-linux4.AppImage --verbose --enable-features=UseOzonePlatform --ozone-platform=wayland
```

I'm also not sure if the AppImage is passing these parameters through to the app - does the same thing occur when you use the `deb` or `rpm` package and invoke the `github-desktop` shortcut?\

EDIT: based on https://github.com/shiftkey/desktop/issues/591#issuecomment-1048239499 it does seem like those parameters should work, but I don't believe `--verbose` is supported.

**Comment by red5h4d0w** [comment]
Hi it was working fine on 2.99 as said in #591 (I'm the same person that opened that issue), but now it can't launch using the same args. 
I added `--verbose` to get a better log of what happened in case that would help.

**Comment by shiftkey** [comment]
> Hi it was working fine on 2.99 as said in #591 (I'm the same person that opened that issue), but now it can't launch using the same args.

I'm not sure what's changed here as these are both using Electron `17.0.1`, and even `2.9.6` is using Electron `16` which should be fine also.

I wish I knew more about what this error meant for your setup, as the `kNull` mention here suggests some prerequisite is missing:

```
[11772:0405/161655.685004:ERROR:cursor_loader.cc(116)] Failed to load a platform cursor of type kNull
Erreur de segmentation (core dumped)
```

[This appears to be the relevant source](https://source.chromium.org/chromium/chromium/src/+/refs/tags/98.0.4758.82:ui/aura/cursor/cursor_loader.cc;l=112-117).

**Comment by red5h4d0w** [comment]
I'm not sure what causes this bug, but it seems to also happen in vscodium [this issue](https://github.com/VSCodium/vscodium/issues/1035). I don't know if this can help to pinpoint the error, but I hope so.

**Comment by shiftkey** [comment]
Based on https://github.com/VSCodium/vscodium/issues/1035#issuecomment-1090470357 let's see when the next release is available for Electron 17 - the commit with the fix does not appear to be in any releases https://github.com/electron/electron/commit/db0ad4f58de9724cc923a693b11ac28ca41390ba

EDIT: nevermind, it actually appears to be in [`18.0.1`](https://www.electronjs.org/releases/stable?page=1#18.0.1) and [`17.3.1`](https://www.electronjs.org/releases/stable?page=2#17.3.1). We're on `17.0.1` currently. If you're able to build from source like this you should be able to see if the problem is reproducible after bumping the dependency:

```diff
diff --git a/package.json b/package.json
index 9e1d579fa..e94f446e6 100644
--- a/package.json
+++ b/package.json
@@ -158,7 +158,7 @@
     "@types/webpack-hot-middleware": "^2.25.6",
     "@types/webpack-merge": "^5.0.0",
     "@types/xml2js": "^0.4.0",
-    "electron": "17.0.1",
+    "electron": "17.3.1",
     "electron-builder": "^23.0.3",
     "electron-packager": "^15.4.0",
     "electron-winstaller": "^5.0.0",
```

And then creating the app:

```
$ yarn
$ yarn build:dev
$ yarn start
```

If that works, let me know.

**Comment by shiftkey** [comment]
I've published [`2.9.14-linux2`](https://github.com/shiftkey/desktop/releases/tag/release-2.9.14-linux2) which contains an Electron build with the update, so hopefully this improves the Wayland support on our end...

**Comment by red5h4d0w** [comment]
Thank you. I've just tested the appimage for 2.9.14-linux2 and this issue is fixed
